api/v1alpha1/project_types.go

```diff
@@ -17,10 +17,16 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"fmt"
+
 	"github.com/operator-framework/operator-sdk/pkg/status"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	errorMsg string = " is an immutable field, can't be changed while updating"
```

Comment:
Remove starting space you have already taken care of the formatting at `fmt.Errorf("%s %s", "ProjectTemplateKey", errorMsg)`